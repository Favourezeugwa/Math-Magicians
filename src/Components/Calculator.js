import React, { useState } from 'react';
import calculate from '../Logic/calculate';

function Calculator() {
  const [state, setState] = useState({ total: 0, operation: null, next: null });

  const { total, operation, next } = state;

  const handleClick = (event) => {
    setState((state) => calculate(state, event.target.textContent));
  };

  return (
    <div className="calculator-container">
      <input type="text" className="calculator-screen" value={(total || '') + (operation || '') + (next || '')} />
      <div className="calculator-keys">
        <button type="button" className="btn-lightgrey" onClick={handleClick}>AC</button>
        <button type="button" className="btn-lightgrey" onClick={handleClick}>+/-</button>
        <button type="button" className="btn-lightgrey" onClick={handleClick}>%</button>
        <button type="button" className="btn-orange" onClick={handleClick}>รท</button>
        <button type="button" className="btn-lightgrey" onClick={handleClick}>7</button>
        <button type="button" className="btn-lightgrey" onClick={handleClick}>8</button>
        <button type="button" className="btn-lightgrey" onClick={handleClick}>9</button>
        <button type="button" className="btn-orange" onClick={handleClick}>x</button>
        <button type="button" className="btn-lightgrey" onClick={handleClick}>4</button>
        <button type="button" className="btn-lightgrey" onClick={handleClick}>5</button>
        <button type="button" className="btn-lightgrey" onClick={handleClick}>6</button>
        <button type="button" className="btn-orange" onClick={handleClick}>-</button>
        <button type="button" className="btn-lightgrey" onClick={handleClick}>1</button>
        <button type="button" className="btn-lightgrey" onClick={handleClick}>2</button>
        <button type="button" className="btn-lightgrey" onClick={handleClick}>3</button>
        <button type="button" className="btn-orange" onClick={handleClick}>+</button>
        <button type="button" className="btn-lightgrey item0" onClick={handleClick}>0</button>
        <button type="button" className="btn-lightgrey" onClick={handleClick}>.</button>
        <button type="button" className="btn-orange" onClick={handleClick}>=</button>
      </div>
    </div>
  );
}

export default Calculator;
