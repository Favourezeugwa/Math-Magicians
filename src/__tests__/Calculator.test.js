import renderer from 'react-test-renderer';
import Calculator from '../Components/Calculator';

it('Test Calculator renders correctly', () => {
  const tree = renderer
    .create(<Calculator />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
