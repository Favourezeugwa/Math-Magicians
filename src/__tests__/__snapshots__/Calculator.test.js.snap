// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Calculator renders correctly 1`] = `
Array [
  <h1
    className="calculator-header"
  >
    Lets do some Maths!
  </h1>,
  <div
    className="calculator-container"
  >
    <input
      className="calculator-screen"
      type="text"
      value=""
    />
    <div
      className="calculator-keys"
    >
      <button
        className="btn-lightgrey"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="btn-lightgrey"
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        className="btn-lightgrey"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="btn-orange"
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
      <button
        className="btn-lightgrey"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="btn-lightgrey"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="btn-lightgrey"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="btn-orange"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
      <button
        className="btn-lightgrey"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="btn-lightgrey"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="btn-lightgrey"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="btn-orange"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
      <button
        className="btn-lightgrey"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="btn-lightgrey"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="btn-lightgrey"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="btn-orange"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
      <button
        className="btn-lightgrey item0"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="btn-lightgrey"
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="btn-orange"
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>,
]
`;
